catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -288,76 +284,28 @@ private File getBackupFile(Metacard metacard) throws IOException {
         return new File(parent, metacardId);
     }
 
-    /**
-     * Removed the temp file extension off of the backed up metacard.
-     *
-     * @param source the backed up metacard file in which the temp file extension
-     *               is to be removed.
-     * @throws IOException
-     */
-    private void removeTempExtension(File source) throws IOException {
-        LOGGER.debug("Removing {} file extension.", TEMP_FILE_EXTENSION);
-        File destination = new File(StringUtils.removeEnd(source.getAbsolutePath(),
-                TEMP_FILE_EXTENSION));
-        boolean success = source.renameTo(destination);
-        if (success) {
-            LOGGER.debug("Moved {} to {}.",
-                    source.getAbsolutePath(),
-                    destination.getAbsolutePath());
-        } else {
-            LOGGER.debug("Failed to move {} to {}.",
-                    source.getAbsolutePath(),
-                    destination.getAbsolutePath());
-        }
-    }
-
-    /**
-     * Gets an exception message
-     *
-     * @param previousExceptionMessage the previous exception message that we wish to append to
-     * @param metacardsIdsInError      the metacard IDs that processed in error
-     * @param operation                the operation being performed
-     * @return the updated exception message
-     */
-    private String getExceptionMessage(String responseType, String previousExceptionMessage,
-            List<String> metacardsIdsInError, OPERATION operation) {
-        StringBuilder builder = new StringBuilder();
-
-        if (StringUtils.isNotBlank(previousExceptionMessage)) {
-            builder.append(previousExceptionMessage);
-            builder.append(" ");
-        }
+    private void validateDirectory(File directory) {
 
-        builder.append("Error processing ");
-        builder.append(responseType);
-        builder.append(". ");
-
-        switch (operation) {
-        case CREATE:
-            builder.append("Unable to backup metacard(s) [");
-            builder.append(StringUtils.join(metacardsIdsInError, ","));
-            builder.append("]. ");
-            break;
-        case DELETE:
-            builder.append("Unable to delete metacard(s) [");
-            builder.append(StringUtils.join(metacardsIdsInError, ","));
-            builder.append("] from backup. ");
-            break;
-        default:
-            LOGGER.debug("No specific error message details for operation {}", operation);
+        if (!(directory.exists() && directory.canWrite())) {
+            throw new IllegalArgumentException("Directory " + directory.getAbsolutePath()
+                    + " does not exist or is not writable");
         }
-        return builder.toString();
     }
 
-    private enum OPERATION {
-        CREATE,
-        DELETE
+    @SuppressWarnings("unused")
+    public long getTerminationTimeoutSeconds() {
+        return terminationTimeoutSeconds;
     }
 
+    @SuppressWarnings("unused")
```

Comment:
Why do we need to set the termination timeout if we're mocking the executor service?

If we really need this and it is only for unit testing purposes, we should make the method package private and put a comment on the method.